modin/core/dataframe/pandas/utils.py

```diff
@@ -65,3 +67,56 @@ def concatenate(dfs):
         # so do it explicitly
         return dfs[0].copy()
     return pandas.concat(dfs, copy=True)
+
+
+def create_dataframe_from_partition_data(partition_data, partition_shape):
+    """
+    Convert partition data of multiple dataframes to a single dataframe.
+
+    Parameters
+    ----------
+    partition_data : list
+        List of pandas DataFrames/Object references holding pandas DataFrames.
+    partition_shape : tuple
+        Shape of the partitions numpy array.
+
+    Returns
+    -------
+    pandas.DataFrame
+        A pandas DataFrame.
+    """
+    if all(
+        isinstance(obj, (pandas.DataFrame, pandas.Series)) for obj in partition_data
+    ):
+        height, width, *_ = tuple(partition_shape) + (0,)
+        # restore 2d array
+        objs = iter(partition_data)
+        partition_data = [[next(objs) for _ in range(width)] for __ in range(height)]
+    else:
+        # Partitions do not always contain pandas objects, for example, hdk uses pyarrow tables.
+        # This implementation comes from the fact that calling `partition.get`
+        # function is not always equivalent to `partition.to_pandas`.
+        partition_data = [[obj.to_pandas() for obj in part] for part in partition_data]
+    if all(isinstance(part, pandas.Series) for row in partition_data for part in row):
+        axis = 0
+    elif all(
+        isinstance(part, pandas.DataFrame) for row in partition_data for part in row
+    ):
+        axis = 1
+    else:
+        ErrorMessage.catch_bugs_and_request_email(True)
+
+    def is_part_empty(part):
+        return part.empty and (
+            not isinstance(part, pandas.DataFrame) or (len(part.columns) == 0)
+        )
+
+    df_rows = [
+        pandas.concat([part for part in row], axis=axis, copy=False)
+        for row in partition_data
+        if not all(is_part_empty(part) for part in row)
+    ]
+    if len(df_rows) == 0:
+        return pandas.DataFrame()
+    else:
+        return concatenate(df_rows)
```

Comment:
Let's add a parameter named `copy`, which we would pass in pandas.concat. We could avoid a data copy in a remote function because Ray itself will store a copy of data.